### DISCUSSION TITLE
failed to initialize database, got error Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub

### DISCUSSION THREAD

**Discussion by donhuvy** [go-gorm/gorm#6095] [Q&A]
I am learning gorm, with authoritative document at https://gorm.io/docs/index.html

File `vy_sqlite.go`

```go
package main

import (
	"gorm.io/driver/sqlite"
	"gorm.io/gorm"
)

type Product struct {
	gorm.Model
	Code  string
	Price uint
}

func main() {
	db, err := gorm.Open(sqlite.Open("test.db"), &gorm.Config{})
	if err != nil {
		panic("failed to connect database")
	}

	// Migrate the schema
	db.AutoMigrate(&Product{})

	// Create
	db.Create(&Product{Code: "D42", Price: 100})

	// Read
	var product Product
	db.First(&product, 1)                 // find product with integer primary key
	db.First(&product, "code = ?", "D42") // find product with code D42

	// Update - update product's price to 200
	db.Model(&product).Update("Price", 200)
	// Update - update multiple fields
	db.Model(&product).Updates(Product{Price: 200, Code: "F42"}) // non-zero fields
	db.Model(&product).Updates(map[string]interface{}{"Price": 200, "Code": "F42"})

	// Delete - delete product
	db.Delete(&product, 1)
}
```

log

```go
API server listening at: 127.0.0.1:62919
WARNING: undefined behavior - version of Delve is too old for Go version 1.20.1 (maximum supported version 1.19)

2023/02/23 16:51:11 C:/Users/admin/GolandProjects/awesomeProject1/vy_sqlite.go:15
[error] failed to initialize database, got error Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub
```
[![enter image description here][1]][1]

How to fix it?

https://stackoverflow.com/questions/75543150/failed-to-initialize-database-got-error-binary-was-compiled-with-cgo-enabled-0

  [1]: https://i.stack.imgur.com/3nDST.png

**Comment by DilyaraKaz** [comment]
where answeer beach

**Comment by Frier03** [comment]
Set the `CGO_ENABLED` to 1. `go-sqlite3` requires cgo to work!

You can set the variable using `$env:CGO_ENABLED=1`

**Reply by JoyCood** [reply]
i set the CGO_ENABLED=1, but it's still not work.

**Reply by nccapo** [reply]
i set CGO_ENABLED to 1, but still have same issue

**Reply by zhangyiming748** [reply]
me too

**Reply by clabnet** [reply]
me too

**Comment by matiasdev30** [comment]
sames issues here

**Comment by zhangyiming748** [comment]
install `build-essential` and `libsqlite3-dev`
```bash
sudo apt-get install build-essential libsqlite3-dev
```
This is how I solved it

**Comment by nccapo** [comment]
First of all check CGO_ENABLED value: `go env CGO_ENABLED`
Then try to update it using: `go env -w CGO_ENABLED=<0 or1>`

Source: https://forum.golangbridge.org/t/getting-errors-when-running-go-project-that-uses-github-com-mattn-go-sqlite3-library/31800

**Reply by zhangyiming748** [reply]
thanks

**Comment by ngshiheng** [comment]
If anyone is currently working with Dockerfile and has encountered a similar issue, I found that adding the following lines to my Dockerfile resolved the problem ([ref](https://github.com/ngshiheng/michelin-my-maps/commit/44de75380c5bce53828d4c1968a3bfb9c0bfecf5)):

```Dockerfile
FROM golang:1.21-alpine AS go-builder
ENV CGO_ENABLED=1
RUN apk add --no-cache gcc musl-dev
RUN go build -ldflags='-s -w -extldflags "-static"' -o /bin/mym ./cmd/mym
# ...
```

I found this solution by referring to the [Dockerfile](https://github.com/mattn/go-sqlite3/blob/6ee3e6746e6b5a0f4099d0553120ceead9f9fc38/_example/simple/Dockerfile) in the `go-sqlite3` repo

**Reply by dkrest1** [reply]
it took so long to build

**Reply by dkrest1** [reply]
but it work

**Comment by Chenorlive** [comment]
hello

if yow are on window see this error 
     Install gcc complier
    add to environment variable  
    restart your terminal 
    go run
    
   if not okay 
   run $env:CGO_ENABLED=1
  go run .
